Makefile

```diff
@@ -1,103 +1,7 @@
-
-build:
-	@echo "Building images..."
-	@docker-compose -f docker-compose.yaml -f docker-compose-ui.yaml -f docker-compose-tests.yaml -f docker-compose-ui-tests.yaml build
-
-up:
-	@docker-compose -f docker-compose.yaml -f docker-compose-ui.yaml up -d postgres graphql-engine admin-ui
-	@chmod u+x ./backend/database-service/scripts/waitForService.sh
-	@./backend/database-service/scripts/waitForService.sh localhost ${API_PORT}
-
-down:
-	@docker-compose down --remove-orphans
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml down
-
-rm:
-	@docker-compose rm -f
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml rm -f
-
-db.migrate.apply:
-	$(HASURA_MIGRATE_APPLY) --project backend/database-service --skip-update-check
-
-db.migrate.status:
-	$(HASURA_MIGRATE_STATUS) --project backend/database-service --skip-update-check
-
-fixtures.generate:
-	@echo "Generating fixtures ..."
-	@chmod u+x ./backend/fixtures-generator/entrypoint.sh
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml up fixtures-service
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml rm -f fixtures-service
-
-fixtures.up:
-	$(HASURA_MIGRATE_APPLY) --project $(FIXTURES_FOLDER)/database/small --up all --skip-update-check
-
-fixtures.down:
-	# TODO: get the number of migrations from a truncated $(shell echo $((`ls database-service/migrations | wc -l`/2)))
-	$(HASURA_MIGRATE_APPLY) --project $(FIXTURES_FOLDER)/database/small --down 6 --skip-update-check
-
-fixtures.clean:
-	rm -rf $(FIXTURES_FOLDER)
-	rm -rf $(GRAPHQL_RESPONSES_FOLDER)
-
-fixtures: fixtures.clean fixtures.generate fixtures.up
-
-test.database-service:
-	@chmod u+x ./backend/database-service/tests/entrypoint.sh
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml up --abort-on-container-exit hasura-service-tests
-
-test.ui-unit:
-	@chmod u+x ./frontend/tests/entrypoint-unit.sh
-	@docker-compose -f docker-compose.yaml -f docker-compose-ui.yaml -f docker-compose-ui-tests.yaml up --abort-on-container-exit ui-unit-tests
-
-test.ui-integration:
-	@chmod u+x ./frontend/tests/entrypoint-integration.sh
-	@docker-compose -f docker-compose.yaml -f docker-compose-ui.yaml -f docker-compose-ui-tests.yaml up --abort-on-container-exit ui-integration-tests
-
-test.e2e:
-	@chmod u+x ./frontend/tests/entrypoint-e2e.sh
-	@docker-compose -f docker-compose.yaml -f docker-compose-ui.yaml -f docker-compose-ui-tests.yaml up --abort-on-container-exit e2e-tests
-
-test.behave:
-	@docker-compose -f docker-compose.yaml -f docker-compose-tests.yaml up --abort-on-container-exit features-tests
-
-backend.test: test.database-service
-ui.test: test.ui-unit test.ui-integration test.e2e
-test: backend.test ui.test
-
-%.restart:
-	make $*.down
-	make $*.up
-
-logs:
-	docker-compose logs -f
-
-console:
-	hasura console --project backend/database-service
-
-%.logs:
-	docker-compose logs -f $*
+%:
```

Comment:
So ... a `make up` in the "root Makefile" triggers a `make up` in the `backend` dir as well as in the `frontend` dir? Isn't that a little bit "dangerous" and too "powerful" ?

Reply:
We will see. Currently, it makes life easier.